### PR TITLE
Updating cdfPlot.R with cx plot functionality

### PR DESCRIPTION
**Pull request by kar-agg-gen**
- Added Cx plot to cdfPlot.R
- Updated unit-test to check Cx plot.
- Updated documentation.

### CODE REVIEW

File: R/cdfPlot.R [hunk 1]
```diff
@@ -113,6 +113,8 @@ cdfPlot <- function(contrastDF,
 
     assertthat::assert_that(pvalCol %in% colnames(contrastDF),
                             msg = "Specified pvalCol not found in the supplied dataframe (contrastDF).")
+    assertthat::assert_that(plotType %in% c("ggplot", "canvasXpress"),
+                            msg = "Plot type must be either ggplot or canvasXpress.")
```
Comment (cb4ds) [inline R/cdfPlot.R:117]: since default is CX list that option first.

File: R/cdfPlot.R [hunk 2]
```diff
@@ -83,9 +82,11 @@
...
 #' @importFrom dplyr arrange left_join
 #' @importFrom assertthat assert_that
+#' @importFrom canvasXpress canvasXpress
 #'
 #' @export
 cdfPlot <- function(contrastDF,
+                    plotType = "canvasXpress",
                     pvalCol = "P.Value",
                     pThreshold = 0.01,
                     xlab = NULL, ylab = NULL,
```
Comment (cb4ds) [inline R/cdfPlot.R:92]: cleanup formatting, no two parameters on a single line please

File: R/cdfPlot.R [hunk 3]
```diff
@@ -171,117 +173,154 @@ cdfPlot <- function(contrastDF,
...
-
-    # Optional Decorations
-    if (!is.null(referenceLine)) {
-        cdfMain <- cdfMain +
-            geom_hline(yintercept = pThreshold, color = referenceLine,
-                       size = refLineThickness, alpha = 0.5)
-    }
+    if (plotType == "canvasXpress") {
+        ## Create the canvasXpress df and var annotation
+        # Main plot
```
Comment (cb4ds) [inline R/cdfPlot.R:178]: comments seem incongruent, what is happening here - creating the df/var or the main plot ?  Comments should add to the user's reading of the code when it isn't clear

File: R/cdfPlot.R [hunk 4]
```diff
@@ -171,117 +173,154 @@ cdfPlot <- function(contrastDF,
...
-        print(cdfInset, vp = vp)
-    }
+        # A viewport taking up a fraction of the plot area (upper left)
+        vp <- grid::viewport(width = viewportWidth, height = viewportWidth,
+                             x = viewportX, y = vy,
+                             just = c("left", "top"))
 
-    MyList <- list(main = cdfMain, inset = cdfInset, viewport = vp)
-    return(MyList)
+        if (printPlot == TRUE) {
```
Comment (cb4ds) [inline R/cdfPlot.R:318]: no need for == TRUE : if (printPlot) ...

**Review by cb4ds — CHANGES_REQUESTED** [review]

File: R/cdfPlot.R (hunk 1 above)
Comment (kar-agg-gen) [reply R/cdfPlot.R]: Done

File: R/cdfPlot.R (hunk 2 above)
Comment (kar-agg-gen) [reply R/cdfPlot.R]: Done

File: R/cdfPlot.R (hunk 3 above)
Comment (kar-agg-gen) [reply R/cdfPlot.R:178]: I have updated the comment.

File: R/cdfPlot.R (hunk 4 above)
Comment (kar-agg-gen) [reply R/cdfPlot.R]: Updated

**Review by cb4ds — APPROVED** [review]

**Pull request merged**